### PR TITLE
Updates to allow for other mepo styles

### PR DESCRIPTION
**Pull request by mathomp4**
This PR adds preliminary support for the three mepo styles: `foo`, `foo@`, `@foo`. Setting as a draft for now as I know @tclune is working on this as well.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,17 @@ set (DOING_GEOS5 YES)
...
 set (FV_PRECISION "R4R8" CACHE STRING "Precision of FV3 core (R4, R4R8, R8)")
 
-list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/@cmake")
+# mepo can now clone subrepos in three styles
+set (ESMA_CMAKE_DIRS
+  cmake
+  @cmake
+  cmake@
+  )
+foreach (dir IN LISTS ESMA_CMAKE_DIRS)
```
Comment (tclune) [inline CMakeLists.txt:36]:
```suggestion
# mepo can now clone subrepos in three styles
foreach (dir cmake @cmake cmake@)
```
Is much shorter and arguably clearer.

File: parallel_build.csh [hunk 2]
```diff
@@ -30,7 +30,7 @@ while ($#argv)
    shift
 end
 
-if (! -d ${ESMADIR}/@env) then
+if (! -d ${ESMADIR}/@env || ! -d ${ESMADIR}/env@ || ! -d ${ESMADIR}/env) then
```
Comment (tclune) [inline parallel_build.csh:33]: My PR will include a macro which can change this to:
```cmake
esma_mepo_style(env mepo_env)
```
And by this point, the macro is available.   One then uses `${mepo_env}` below to eliminate the `if` chain.

File: parallel_build.csh [hunk 3]
```diff
@@ -57,5 +57,9 @@ set argv = "$origargv"
 
 if ( -d ${ESMADIR}/@env ) then
    ${ESMADIR}/@env/build.csh -esmadir $ESMADIR $argv
+else if ( -d ${ESMADIR}/env@ ) then
+   ${ESMADIR}/env@/build.csh -esmadir $ESMADIR $argv
+else if ( -d ${ESMADIR}/env ) then
+   ${ESMADIR}/env/build.csh -esmadir $ESMADIR $argv
```
Comment (tclune) [inline parallel_build.csh:63]: I missed this bit on my laptop since I don't use env.

**Review by tclune — COMMENTED** [review]
Since my changes are (almost) a superset of these, should I just do a separate PR or should I try to push to your feature branch?

File: parallel_build.csh (hunk 2 above)
Comment (mathomp4) [reply parallel_build.csh:33]: I don't think your macro will work here. This is a C-shell script and there is not really a prettier way to fix this.

**Review by tclune — APPROVED** [review]

File: parallel_build.csh (hunk 2 above)
Comment (mathomp4) [reply parallel_build.csh:33]: Well, I mean, I could make a pseudo-csh function but...eh. I'd rather rewrite in Bash at that point!

**Comment by mathomp4** [comment]
Closing in favor of #334

**Pull request closed without merging**